build-tools/packages/build-cli/docs/publish.md

```diff
@@ -0,0 +1,43 @@
+`flub publish`
+==============
+
+Publish commands are used to publish packages to an npm registry.
+
+* [`flub publish tarballs`](#flub-publish-tarballs)
+
+## `flub publish tarballs`
+
+Publishes tarballs to the package registry unless the version is already published.
+
+```
+USAGE
+  $ flub publish tarballs [-v | --quiet] [--dir <value>] (--tarball --orderFile <value>) [--retry <value>] [--dryRun]
+    [--access public|restricted] [--publishArgs <value>]
+
+FLAGS
+  --access=<option>      This flag will be passed to 'npm publish'.
+                         <options: public|restricted>
+  --dir=<value>          [default: .] A directory containing tarballs to publish. Tarballs must have the file extension
+                         tgz.
+  --dryRun               Does everything except publish to the registry. This flag will be passed to 'npm publish'.
+  --orderFile=<value>    (required) A file with package names that should be published. Such files can be created using
+                         `flub list`.
+  --publishArgs=<value>  This string will be passed to 'npm publish' verbatim. Use this to pass additional custom args
+                         to npm publish like --tag.
+  --retry=<value>        Number of times to retry publishing a package that fails to publish.
+  --tarball              Use this flag to indicate that the orderFile contains tarball names instead of package names.
+                         Such files can be created using `flub list --tarball`. This option is deprecated and for
+                         backwards compatibility only.
```

Comment:
No way to avoid adding something deprecated from the get go?

Reply:
Unfortunately not really. The current pipeline creates files using the tarball names. I want to switch that to use package names, but I want to make those changes separately. That is, I want to first remove the old publishing logic and replace it with this, then separately update the package enumeration logic once the publishing is working as expected.